client/ayon_deadline/plugins/publish/global/collect_jobinfo.py

```diff
@@ -0,0 +1,347 @@
+# -*- coding: utf-8 -*-
+import json
+
+import pyblish.api
+from ayon_core.lib import (
+    BoolDef,
+    NumberDef,
+    EnumDef,
+    TextDef,
+    UISeparatorDef
+)
+from ayon_core.pipeline.publish import AYONPyblishPluginMixin
+from ayon_core.lib.profiles_filtering import filter_profiles
+from ayon_core.addon import AddonsManager
+
+from ayon_deadline.lib import (
+    FARM_FAMILIES,
+    AYONDeadlineJobInfo,
+)
+
+
+class CollectJobInfo(pyblish.api.InstancePlugin, AYONPyblishPluginMixin):
+    """Collect variables that belong to Deadline's JobInfo.
+
+    Variables like:
+    - department
+    - priority
+    - chunk size
+
+    """
+
+    order = pyblish.api.CollectorOrder + 0.420
+    label = "Collect Deadline JobInfo"
+
+    families = FARM_FAMILIES
+    targets = ["local"]
+
+    profiles = []
+    pool_enum_values = []
+    group_enum_values = []
+    limit_group_enum_values = []
+    machines_enum_values = []
+
+    def process(self, instance):
+        if not instance.data.get("farm"):
+            self.log.debug("Should not be processed on farm, skipping.")
+            return
+
+        attr_values = self._get_jobinfo_defaults(instance)
+
+        attr_values.update(self.get_attr_values_from_data(instance.data))
+        job_info = AYONDeadlineJobInfo.from_dict(attr_values)
+
+        self._handle_machine_list(attr_values, job_info)
+
+        self._handle_additional_jobinfo(attr_values, job_info)
+
+        instance.data["deadline"]["job_info"] = job_info
+
+        # pass through explicitly key and values for PluginInfo
+        plugin_info_data = None
+        if attr_values["additional_plugin_info"]:
+            plugin_info_data = (
+                json.loads(attr_values["additional_plugin_info"]))
+        instance.data["deadline"]["plugin_info_data"] = plugin_info_data
+
+    def _handle_additional_jobinfo(self,attr_values, job_info):
+        """Adds not explicitly implemented fields by values from Settings."""
+        additional_job_info = attr_values["additional_job_info"]
+        if not additional_job_info:
+            return
+        for key, value in json.loads(additional_job_info).items():
+            setattr(job_info, key, value)
+
+    def _handle_machine_list(self, attr_values, job_info):
+        machine_list = attr_values["machine_list"]
+        if machine_list:
+            if attr_values["machine_list_deny"]:
+                job_info.Blacklist = ",".join(machine_list)
+            else:
+                job_info.Whitelist = ",".join(machine_list)
+
+    @classmethod
+    def apply_settings(cls, project_settings):
+        settings = project_settings["deadline"]
+        profiles = settings["publish"][cls.__name__]["profiles"]
+
+        cls.profiles = profiles or []
+
+        addons_manager = AddonsManager()
```

Comment:
I strongly believe this should not happen during `apply_settings` but first time any of those values is needed.